EIPS/eip-1559.md

```diff
@@ -209,6 +209,13 @@ class World(ABC):
 
 			# ensure that the user was willing to at least pay the base fee
 			assert transaction.max_fee_per_gas >= block.base_fee_per_gas
+
+			# The first two of these four rules are implicit due to the next two rules
+			assert transaction.max_fee_per_gas < 2**256           # Prevent impossibly large numbers
+			assert transaction.max_priority_fee_per_gas < 2**256  # Prevent impossibly large numbers
+			assert transaction.max_fee_per_gas >= transaction.max_priority_fee_per_gas      # The total must be the larger of the two
+			assert sender.balance >= gasLimit * transaction.max_fee_per_gas
```

Comment:
sender or signer?

Reply:
`signer`.  That is my bad, I didn't expect my hacked together sample code in Discord to make it into the spec.  😜

Reply:
This is weird when you specify the expected value for an assert on the left side of a logical operator.

Reply:
I consider the right side to be the expected value here.  We are checking to make sure the sender has enough money to cover costs, so the "sender's balance" is the actual, and `gasLimit * transaction.max_fee_per_gas` is the expected.